logging/src/main/scala/com/ovoenergy/effect/Logging.scala

```diff
@@ -0,0 +1,75 @@
+package com.ovoenergy.effect
+
+import cats.data.StateT
+import cats.effect.Sync
+import cats.syntax.flatMap._
+import cats.{Applicative, FlatMap}
+import com.typesafe.scalalogging.LazyLogging
+import org.slf4j.MDC
+
+import scala.language.higherKinds
+
+/**
+  * A type class representing the ability to log
+  * within some effect type F
+  */
+trait Logging[F[_]] {
+  def log(message: Logging.Log): F[Unit]
+}
+
+/**
+  * Our functional code doesn't want to be sprinkled with effectful log statements
+  * and it also doesn't want to be tied to the IO monad, so we define a type class for some F
+  * which permits logging, returning an unmodified value wrapped in a (presumably modified) F
+  * In production code we can use SLF4J and IO, in tests we can use a Writer[List[Log]]
+  */
+object Logging {
+
+  type Tags = Map[String, String]
+
+  sealed trait Log {
+    def tags: Tags
+  }
+
+  case class Debug(of: String, tags: Tags = Map.empty) extends Log
+  case class Info(of: String, tags: Tags  = Map.empty) extends Log
+  case class Error(of: String, tags: Tags = Map.empty) extends Log
```

Comment:
Having Error to take an `Ior[String, Throwable]` would be a useful thing to have.

Reply:
oh yes, there are multiple versions of this library floating about 😓  - I'll add that as version 1.1!